tests/memberships/test_membership_services.py

```diff
@@ -0,0 +1,176 @@
+def community_creation_input_data():
+    """Full community data used as input to community service."""
+    return  {
+        "access": {
+            "visibility": "public",
+            "member_policy": "open",
+            "record_policy": "open",
+        },
+        "id": "my_community_id",
+        "metadata": {
+            "title": "My Community",
+            "description": "This is an example Community.",
+            "type": "event",
+            "curation_policy": "This is the kind of records we accept.",
+            "website": "https://inveniosoftware.org/",
+            "organizations": [{
+                    "name": "CERN",
+            }]
+        }
+    }
+
+
+@pytest.fixture(scope="function")
+def community_service(app, location):
+    """Community service.
+
+    Snuck in the location fixture, because needed on community creation
+    i.e. almost every time this service is used.
+    """
+    return current_communities.service
+
+
+@pytest.fixture(scope="module")
+def requests_service(app):
+    """Requests service."""
+    return current_requests_service
+
+
+@pytest.fixture(scope="function")
+def community(community_service, community_owner_identity,
+              community_creation_input_data):
+    """Community fixture."""
+    return community_service.create(
+        community_owner_identity, community_creation_input_data
+    )
+
+
+@pytest.fixture()
+def invitation_creation_input_data(another_user, community):
+    """Full invitation data used as input to invitation service."""
+    return {
+        "type": "community-member-invitation",
+        "receiver": {"user": another_user.id},
+        "topic": {'community': community.data['uuid']}
+    }
+
+
+# Tests
+
+def test_invite_user_flow(
+        another_identity, community, community_owner_identity,
+        community_service, invitation_creation_input_data,
+        requests_service):
+    community_id = community.data['uuid']
+
+    # Invite
+    invitation = community_service.invitations.create(
+        community_owner_identity, invitation_creation_input_data
+    )
+
+    invitation_dict = invitation.to_dict()
+    assert 'open' == invitation_dict['status']
+    assert invitation_dict['is_open'] is True
+    assert {'community': community_id} == invitation_dict['topic']
+    assert {'user': str(another_identity.id)} == invitation_dict['receiver']
+    assert {'community': community_id} == invitation_dict['created_by']  # noqa
+
+    # Accept
+    invitation = requests_service.execute_action(
+        another_identity, invitation.id, 'accept', {}
+    )
+
+    invitation_dict = invitation.to_dict()
+    assert 'accepted' == invitation_dict['status']
+    assert invitation_dict['is_open'] is False
+    assert invitation_dict['is_closed'] is True
+
+    # Get membership
+    member = community_service.members.read(
+        community_owner_identity,
+        {
+            "community": community_id,
+            "user": str(another_identity.id),
+        }
```

Comment:
1- In a future PR I will clarify the distinction between community id and community uuid. Here community_id means community uuid. I think at the data layer level we want uuid used and at the resource level, the id. This means the service would have to convert between one and the other. Do we have mechanisms to do so?

2- Because I see the endpoint as: `/api/communities/<community_id>/members/<user|group>/<entity_id>`, the id for the `read` method is a composite id. This hasn't really been the case for services in the past I think. Thoughts?